x/wasm/internal/types/genesis.go

```diff
@@ -0,0 +1,29 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GenesisState is the struct representation of the export genesis
+type GenesisState struct {
```

Comment:
This in general seems very hard to validate. The fact that the various components for any contract are scattered. Using the same information, how about:

```go
type GenesisState []ContractState
// (or type GenesisState map[string]ContractState if that makes more sense)

type ContractState struct {
        CodeInfo         CodeInfo       `json:"code_info"`
	CodesBytes        []byte         `json:"code_bytes"`
	ContractAddress sdk.AccAddress `json:"contract_address"`
	ContractInfo     Contract       `json:"contract_info"`
	ContractState    []Model        `json:"contract_state"`
}
```

Reply:
I thought about this before and have some issues with such a huge code-dump in genesis. For example, the CodeBytes is 60-100kb, maybe this could live in a binary file next to the genesis file, and just keep a reference here. Then again, maybe that violates the idea that the genesis file is everything you need. I just see that a few contracts could easily add a MB to the genesis size. And if there are 100s, this will make the genesis file almost unusable.

Reply:
After reading through the pr more, I realize there are two separate groups - Code and Contract - which need their own indexes. But I would still group all info on one object together. Maybe something like:

```go
type GenesisState struct {
  Code []CodeState
  Contracts []ContractState
}

type CodeState struct {
        CodeInfo         CodeInfo       `json:"code_info"`
	CodesBytes        []byte         `json:"code_bytes"`
}

type ContractState struct {
	ContractAddress sdk.AccAddress `json:"contract_address"`
	ContractInfo     Contract       `json:"contract_info"`
	ContractState    []Model        `json:"contract_state"`
}
```